### PR TITLE
Docs: add examples for `RuntimeEnv::register_object_store`, improve error messages

### LINKED ISSUE #10616: Make it easier to register object stores
Came up in  https://github.com/apache/datafusion/issues/9280 

The problem is that when trying to use urls in ListingTables (or ParquetExec), if you try with a scheme like `file://` it won't work unless you have registered an object store handler

However, the current user experience is bad -- the error message is mysterious and confusing and it is not obvious how to register the correct listing table

"Error: Internal("No suitable object store found for file://")"

I would recommend two things:

# Suggestion 1: Change the error message 

Change the error message to include instructions on how to fix the issue

For example, instead of 
> No suitable object store found for file://

Perhaps it could be something more like

> No object store registered for url scheme 'file://'. See `RuntimeEnv::register_object_store`

# Suggestion 2: Add an example to the docs

Once the error message directs to  `RuntimeEnv::register_object_store` then it would be ideal if the docs

https://docs.rs/datafusion/latest/datafusion/execution/runtime_env/struct.RuntimeEnv.html#method.register_object_store

actually had an example

So that would perhaps take the form of adapting the example above as a doc example

Does that make sense?

_Originally posted by @alamb in https://github.com/apache/datafusion/issues/9280#issuecomment-2123183351_

### PR DESCRIPTION
**Pull request by aditanase**
## Which issue does this PR close?

Closes #10616. Documentation only

### CODE REVIEW

**Comment by alamb** [comment]
Thank you @aditanase  -- this is great 🙏 . I started the CI

Also I made another PR to try and improve the documentation about `ObjectStoreUrl` in general as I found that confusing

File: datafusion/execution/src/runtime_env.rs [hunk 1]
```diff
@@ -89,6 +89,27 @@ impl RuntimeEnv {
...
+    /// ```
+    /// let url = Url::try_from("file://").unwrap();
+    /// let object_store = object_store::local::LocalFileSystem::new();
+    /// ctx.runtime_env()
+    ///    .register_object_store(&url, Arc::new(object_store));
+    /// ```
+    ///
+    /// Or how to set up a http object store:
+    /// ```
+    /// let base_url = Url::parse("https://github.com").unwrap();
```
Comment (alamb) [inline datafusion/execution/src/runtime_env.rs:117]: These examples are run automatically via `cargo test --doc` -- so they need to be completely standlone

I took the liberty of pushing a commit to this PR that did so

**Review by alamb — APPROVED** [review]
Thanks again @aditanase  -- really apprecaited

File: datafusion/execution/src/runtime_env.rs (hunk 1 above)
Comment (aditanase) [reply datafusion/execution/src/runtime_env.rs:117]: Thanks so much for making these pass, I'll remember to run the doc tests next time, only looked at the doc site before submitting the PR.

**Pull request merged**